File: part02/ch03/3.py
n, m = map(int, input().split())
result = 0
for _ in range(n):
    data = list(map(int, input().split()))
    result = max(result, min(data))

print(result)

'''
[input]
3 3
3 1 2
4 1 4
2 2 2
[output]
2

[input]
2 4
7 3 1 8
3 3 3 4
[output]
3
'''

File: part02/ch03/3.cpp
#include <bits/stdc++.h>

using namespace std;

int n, m;
int result = 0;

int main() {
    cin >> n >> m;
    for (int i = 0; i < n; i++) {
        int min_value = 10001;
        for (int j = 0; j < m; j++) {
            int num;
            cin >> num;
            min_value = min(min_value, num);
        }
        result = max(result, min_value);
    }

    cout << result;
}